northstar/src/runtime/mod.rs

```diff
@@ -118,23 +121,20 @@ impl Runtime {
             });
         }
 
-        Ok(Runtime {
-            stop: Some(stop_tx),
-            stopped: stopped_rx,
-            event_tx,
-        })
-    }
-
-    /// Stop the runtime
-    pub fn stop(mut self) {
-        // Drop the sending part of the stop handle
-        self.stop.take();
+        Ok((
+            Runtime {
+                stopped: stopped_rx,
+                event_tx,
+            },
+            token,
+        ))
     }
 
-    /// Stop the runtime and wait for the termination
-    pub fn stop_wait(mut self) -> impl Future<Output = RuntimeResult> {
-        self.stop.take();
-        self
+    /// Wait for the termination
+    pub async fn wait(self) -> RuntimeResult {
```

Comment:
Should me merged into `Runtime::stop`

Reply:
hmm I don't understand why `wait` has to be combined with `stop`.  Sometimes we want to stop without wait and sometimes we want to wait without stop.

Reply:
What's the use case for stopping without waiting for a proper shutdown?

Reply:
For the signal handling for example. They just trigger the stop without waiting.

Reply:
Today there's a wait ;-). I just don't get the point why this is bad.

Reply:
hmmm I didn't understand that :confused: I don't understand why this is bad either

Reply:
Ah got it, yeah the wait is done by someone else. The one owning the runtime is the one waiting for it. That's related to the other discussion of course.